### PR TITLE
refactor(datastore): Convert plugin to use Result instead of AsyncOperation

### PR DESCRIPTION
**Pull request by palpatim**
- chore: Fix SwiftFormat and SwiftLint invocations to use common configs
- chore: Fix linting & formatting issues

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginTests/Sync/RemoteSyncEngineTests.swift [hunk 1]
```diff
@@ -57,6 +59,7 @@ class RemoteSyncEngineTests: XCTestCase {
         remoteSyncEngine.start()
 
         wait(for: [failureOnStorageAdapter], timeout: defaultAsyncWaitTimeout)
+        remoteSyncEngineSink.cancel()
```
Comment (wooj2) [inline AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginTests/Sync/RemoteSyncEngineTests.swift:62]: Just curious, did you run into some problem with remote sync engine needing these cancels?

File: AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginTests/Sync/SubscriptionSync/ModelReconciliationDeleteTests.swift [hunk 2]
```diff
@@ -57,9 +59,9 @@ class ModelReconciliationDeleteTests: SyncEngineTestBase {
         }
         wait(for: [expectationListener], timeout: 2.0)
 
-        guard let listener = listenerFromRequest else {
-            XCTFail("Incoming responder didn't set up listener")
-            return
+        guard let valueListener = valueListenerFromRequest else {
+                XCTFail("Incoming responder didn't set up listeners")
```
Comment (wooj2) [inline AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginTests/Sync/SubscriptionSync/ModelReconciliationDeleteTests.swift:63]: isn't there just one listener? not sure why this was changed from listener to listeners

File: AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/SubscriptionSync/IncomingAsyncSubscriptionEventToAnyModelMapper.swift [hunk 3]
```diff
@@ -45,27 +45,15 @@ final class IncomingAsyncSubscriptionEventToAnyModelMapper: Subscriber, Cancella
...
-        }
+    func receive(_ subscriptionEvent: IncomingAsyncSubscriptionEventPublisher.Event) -> Subscribers.Demand {
+        log.verbose("\(#function): \(subscriptionEvent)")
+        dispose(of: subscriptionEvent)
         return .max(1)
     }
 
     func receive(completion: Subscribers.Completion<DataStoreError>) {
         log.info("Received completion: \(completion)")
+        modelsFromSubscription.send(completion: completion)
```
Comment (wooj2) [inline AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/SubscriptionSync/IncomingAsyncSubscriptionEventToAnyModelMapper.swift:56]: There is a comment on line 65/66 referring to AsyncEvent Completion/Error.  I don't think this is relevant anymore given that we are just receiving `SubscriptionEvent`

File: AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginTests/Sync/RemoteSyncEngineTests.swift (hunk 1 above)
Comment (palpatim) [reply AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginTests/Sync/RemoteSyncEngineTests.swift:62]: No, it's just there to get rid of the "unused variable" compiler warning. (We can't do away with the variable, or else Combine releases the sink before it has a chance to process.)

File: AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginTests/Sync/SubscriptionSync/ModelReconciliationDeleteTests.swift (hunk 2 above)
Comment (palpatim) [reply AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginTests/Sync/SubscriptionSync/ModelReconciliationDeleteTests.swift:63]: Derp. That was there when I thought I needed to `guard` for both the value and completion listeners. Will change back.

File: AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/SubscriptionSync/IncomingAsyncSubscriptionEventToAnyModelMapper.swift (hunk 3 above)
Comment (palpatim) [reply AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/SubscriptionSync/IncomingAsyncSubscriptionEventToAnyModelMapper.swift:56]: Thanks for catching, will change

**Pull request merged**